pkg/core/request/http.go

```diff
@@ -0,0 +1,94 @@
+package request
+
+import (
+	"bytes"
+	"context"
+	"net/http"
+
+	"github.com/ksysoev/wasabi"
+)
+
+type HTTPReq struct {
+	ctx     context.Context
+	headers map[string][]string
+	url     string
+	method  string
+	body    []byte
+	id      int64
+}
+
+// NewHTTPReq creates a new HTTPReq instance with the specified method, URL, and body.
+// It takes three parameters: ctx of type context.Context, method of type string, and url of type string.
+// It also takes body of type []byte which represents the request payload.
+// It returns a pointer to an HTTPReq struct initialized with the provided method, URL, and body.
+func NewHTTPReq(ctx context.Context, method, url string, body []byte, reqID int64) *HTTPReq {
```

Comment:
could we delegate ID generation to the constructor itself? In what cases passing ID to constructor can be risky?

Reply:
@shan-96 id must be uniq within connection, and based on Deriv api it should numeric, that is a I'm keeping responsibility of generating id in connection object... if we would not have such limitation, it'd totally make sense to use uuid and generate it in constructor.